packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -211,6 +215,17 @@ class ArtifactStore {
     return _engineRevision;
   }
 
+  static String _expectedEngineRevision;
+
+  static String get expectedEngineRevision {
+    if (_expectedEngineRevision == null) {
+      File revisionFile = new File(path.join(flutterRoot, 'packages', 'flutter', 'packages', 'sky_engine', 'REVISION'));
```

Comment:
Yeah, re: `.packages` / packagesRoot, we've got at least a dozen places to patch up. The biggest issue will be the snapshot generation tool, which currently only takes a package root arg.